jax_unirep/evotuning.py

```diff
@@ -314,37 +352,41 @@ def step(i, state):
     epoch_len = round(len(sequences) / batch_size)
 
     n = n_epochs * epoch_len
-    for i in range(n):
+    for i in tqdm(range(n), desc="Iteration"):
 
         if i % epoch_len == 0:
             logger.info(f"Starting epoch {int(i / epoch_len) + 1}")
-
-        logging.debug("Getting batches")
+        logger.debug(f"Iteration {i}")
+        logger.debug("Getting batches")
         l = choice(seq_lens)
         x, y = len_batching_funcs[l]()
 
         # actual forward & backwrd pass happens here
-        logging.debug("Getting state")
-        state = step(i, state)
+        logger.debug("Getting state")
+        state = step(i, x, y, state)
 
         if i % epoch_len == 0:
             params = get_params(state)
-            loss = avg_loss(xs, ys, params)
+            x, y = len_batching_funcs[l]()
+            loss = avg_loss([x], [y], params, backend=backend)
```

Comment:
Does this not only use a single batch of sequences instead of all sequences of a given length to calculate the avg loss on the dataset?

Also, is it valid to use just a single length of sequences to calculate avg loss on the dataset when using length batching? In case of random batching, I guess the way we implemented it, it would in any case pass in all sequences because we pad the seqs before we make the `batching_funcs`.

Reply:
I changed the docstring to make it clear that we aren't calculating loss on the entire dataset, we are using a random subsample of size `batch_size` to estimate the loss. It's much too expensive to calculate the loss directly; this change improves our ability to iterate.